### ISSUE TITLE
Unable to validate property that consists of single "$" sign

### ISSUE THREAD

**Issue by magnetnation** [ajv-validator/ajv#87]
If I use the following schema:

```
var schema = {
    "$schema": "http://json-schema.org/draft-04/schema#",
    "type": "object",
    "additionalProperties": false,
    "properties": 
    {
        "$": 
        {
            "type": "string"
        }
    },
    "required": 
    [
        "$"
    ]
};
```

and try to validate this object: 

```
{$: 'Client'}
```

It will fail saying that "Invalid: data.$ should be string".

If I change $ to for example $a validation run without errors.

**Comment by epoberezkin** [comment]
Fixed in version 2.2.2. Thanks!

**Comment by epoberezkin** [comment]
btw, it also affected all properties that contain [replacement patterns](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#Specifying_a_string_as_a_parameter) or end with "$".
